lib/src/shelf_run.dart

```diff
@@ -38,26 +40,46 @@ Future<ShelfRunContext> shelfRun(
 
   if (useHotReload) {
     withHotreload(() async {
-      final server = await _createServer(
+      try {
+        final server = await _createServer(
+          init,
+          defaultBindPort: defaultBindPort,
+          defaultBindAddress: defaultBindAddress,
+          defaultShared: defaultShared,
+          securityContext: securityContext,
+          onStarted: onStarted,
+        );
+        context._server = server;
+      }
+      on SocketException catch(e){
+        if(onStartFailed != null) {
```

Comment:
This code part seems to double. What about simplifying this by adding something like
```
onStartFailed ?= print;
```
some lines above (e.g. line 40)